### ISSUE TITLE
`test $a == $b` should raise SC3014 for posix sh, but does not.

### ISSUE THREAD

**Issue by coolaj86** [koalaman/shellcheck#2857]
#### For bugs
- Rule Id `SC3014`
- My shellcheck version (`shellcheck --version` or "online"): 
   ```sh
   ShellCheck - shell script analysis tool
   version: 0.9.0
   license: GNU General Public License, version 3
   website: https://www.shellcheck.net
   ```
- [x] The rule's wiki page (https://www.shellcheck.net/wiki/SC3014) does not already cover this
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

a=a
b=b

# False Negative - Does NOT raise SC3014
if test $a == $b; then
  echo "snuck right past ya"
fi

# Correctly raises SC3014
if [ $a == $b ]; then
  echo "oh, you got me!"
fi
```

#### Here's what shellcheck currently says:

Nothing doesn't catch the error.

#### Here's what I wanted or expected to see:

Should raise SC3014

**Comment by Ordoviz** [comment]
Probably fixed in #2837
